src/compiler/checker.ts

```diff
@@ -15835,7 +15835,7 @@ namespace ts {
             const callSignatures = getSignaturesOfType(expressionType, SignatureKind.Call);
             if (callSignatures.length) {
                 const signature = resolveCall(node, callSignatures, candidatesOutArray);
-                if (getReturnTypeOfSignature(signature) !== voidType) {
+                if (!isInJavaScriptFile(signature.declaration) && getReturnTypeOfSignature(signature) !== voidType) {

```

Comment:
Probably, but I need to #14056 first to properly resolve the underlying function to get to the symbols. Right now the symbol points to the var declaration from the call to `require`, and doesn't have this information.